Add code of conduct (#18802)
matplotlib/matplotlib: CODE_OF_CONDUCT.md

# Contributor Covenant Code of Conduct

## Our Pledge

We as members, contributors, and leaders pledge to make participation in our
community a harassment-free experience for everyone, regardless of age, body
size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, religion, or sexual identity
and orientation.

We pledge to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## Our Standards

Examples of behavior that contributes to a positive environment for our
community include:

* Demonstrating empathy and kindness toward other people
* Being respectful of differing opinions, viewpoints, and experiences
* Giving and gracefully accepting constructive feedback
* Accepting responsibility and apologizing to those affected by our mistakes,
  and learning from the experience
* Focusing on what is best not just for us as individuals, but for the
  overall community

> Examples of unacceptable behavior include:
> 
> * The use of sexualized language or imagery, and sexual attention or
>   advances of any kind
> * Trolling, insulting or derogatory comments, and personal or political attacks
> * Public or private harassment
> * Publishing others' private information, such as a physical or email
>   address, without their explicit permission
> * Other conduct which could reasonably be considered inappropriate in a
>   professional setting

Reviewer: Suggest we don't end a list of examples with a catch all:

```suggestion
Unacceptable behavior is any behavior that would be considered inappropriate in a professional setting; for example:

* The use of sexualized language or imagery, and sexual attention or
  advances of any kind
* Trolling, insulting or derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or email
  address, without their explicit permission
```

Author: We decided on the call not to try and word-smith the CoC, and just adopt a pre-existing one wholesale.

Reviewer: OK, then just merge it?

Reviewer: Marking unresolved so that other folks can see that they aren't to make suggestions....

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at
matplotlib@numfocus.org (monitored by Thomas Caswell (\@tacaswell) and Hannah
Aizenman (\@story645)), senior NumFOCUS staff at conduct@numfocus.org, or a
report can be made using the [NumFOCUS Code of Conduct report form][numfocus form].

> All complaints will be reviewed and investigated promptly and fairly.

Reviewer 2: Here is where senior NumFOCUS staff at conduct@numfocus.org comes in
> If community leaders cannot come to a resolution about enforcement, reports may (will?) be escalated to the NumFocus Code of Conduct committee (conduct@numfocus.org).

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at
matplotlib@numfocus.org (monitored by Thomas Caswell and Hannah Aizenman), or

> senior NumFOCUS staff at conduct@numfocus.org.

Reviewer 2: the NumFocus anonymous report form is  https://numfocus.typeform.com/to/ynjGdT

 conduct@numfocus.org is the moderator that issues the council cwe don't have can't resolve goes to & the important bit there is probably this link https://numfocus.org/code-of-conduct since it has the people on their CoC board. 

Also Thomas Caswell [@tacaswell] & Hannah Aizenman [@story645 ]

Author: I'm not sure I understood everything above, but is this now better/fixed?

Reviewer 2: Reports get escalated to senior NumFOCUS staff at conduct@numfocus.org if they can't be resolved, but initial reporting should either be to matplotlib@numfocus.org or the typeform.

Reviewer 3: I think it is OK if people want to email conduct@numfocus.org first in cases where I am the problem or they don't trust the matplotlib@numfocus.org address for what ever reason.
